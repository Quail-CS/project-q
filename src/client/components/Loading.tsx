/**
 * @module Loading.tsx
 * @description Placeholder Loading Component
 */

import * as React from 'react';

const Loading = () => {
  return (
    <h1>Loading...</h1>
  )
};

export default Loading;
