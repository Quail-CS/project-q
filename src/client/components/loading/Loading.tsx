/**
 * @module Loading.tsx
 * @description Placeholder Loading Component
 */

import * as React from 'react';
import './Loading.scss';

const Loading = () => {
  return (
    <div className="loader-container">
      <div className="loader">
        <div className="loader__bar"></div>
        <div className="loader__bar"></div>
        <div className="loader__bar"></div>
        <div className="loader__bar"></div>
        <div className="loader__bar"></div>
        <div className="loader__ball"></div>
    </div>
    <h1>Loading</h1>
  </div>
  )
};

export default Loading;
