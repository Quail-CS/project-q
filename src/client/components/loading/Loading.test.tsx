import * as React from 'react';
import * as Enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import * as Renderer from 'react-test-renderer';

import Loading from './Loading';

Enzyme.configure({ adapter: new Adapter() });

describe('Loading Component: ', () => {
  it('matches the snapshot', () => {
    const tree = Renderer.create(<Loading />).toJSON();
    expect(tree).toMatchSnapshot();
  })
  it('renders a <div> with class .loading-container', () => {
    const loadingWrapper: Enzyme.ShallowWrapper = Enzyme.shallow(<Loading />);
    expect(loadingWrapper.type()).toEqual('div');
    expect(loadingWrapper.hasClass('loading-container')).toEqual(true);
  })
  it('contains a <div> with class .loading and an h1', () => {
    const loadingWrapper: Enzyme.ShallowWrapper = Enzyme.shallow(<Loading />);
    const loadingDiv: Enzyme.ShallowWrapper = loadingWrapper.find('.loading');
    const loadingH1: Enzyme.ShallowWrapper = loadingWrapper.find('h1');
    expect(loadingDiv).toHaveLength(1);
    expect(loadingDiv.type()).toEqual('div');
    expect(loadingH1).toHaveLength(1);
  })
  it('the loadingContainer contains 5 loading bars <div>s and a loading ball <div>', () => {
    const loadingWrapper: Enzyme.ShallowWrapper = Enzyme.shallow(<Loading />);
    const loadingContainer: Enzyme.ShallowWrapper = loadingWrapper.find('.loading-container');
    expect(loadingContainer.find('.loading__bar')).toHaveLength(5);
    expect(loadingContainer.find('.loading__bar').first().type()).toEqual('div');
    expect(loadingContainer.find('.loading__ball')).toHaveLength(1);
    expect(loadingContainer.find('.loading__ball').type()).toEqual('div');
  })
  it('the H1 has text "Loading" by default', () => {
    const loadingWrapper: Enzyme.ShallowWrapper = Enzyme.shallow(<Loading />);
    const loadingH1Text: string = loadingWrapper.find('h1').text();
    expect(loadingH1Text).toEqual('Loading');
  })
  it('the H1 has text takes whatever text is passed in the loadingMessage prop ', () => {
    const loadingWrapper: Enzyme.ShallowWrapper = Enzyme.shallow(<Loading loadingMessage="Calculating" />);
    const loadingH1Text: string = loadingWrapper.find('h1').text();
    expect(loadingH1Text).toEqual('Calculating');
  })
});