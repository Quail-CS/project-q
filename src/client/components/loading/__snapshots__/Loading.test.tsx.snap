// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Component:  matches the snapshot 1`] = `
<div
  className="loader-container"
>
  <div
    className="loader"
  >
    <div
      className="loader__bar"
    />
    <div
      className="loader__bar"
    />
    <div
      className="loader__bar"
    />
    <div
      className="loader__bar"
    />
    <div
      className="loader__bar"
    />
    <div
      className="loader__ball"
    />
  </div>
  <h1>
    Loading
  </h1>
</div>
`;
